cli/src/pcluster/templates/pcluster_dashboard_builder.py

```diff
@@ -0,0 +1,586 @@
+            {"metrics": ["PermittedThroughput"], "title": "Permitted Throughput"},
+            {"metrics": ["DataReadIOBytes", "DataWriteIOBytes"], "title": "Data Read/Write IO Bytes"},
+        ]
+        widgets_list = []
+        widgets_list += self._add_storage_widgets(
+            efs_metrics,
+            efs_volumes_list,
+            "AWS/EFS",
+            "FileSystemId",
+            "file_system_id",
+        )
+
+        # Conditional EFS metrics
+        conditional_efs_metrics_params = [
+            {
+                "metric": "PercentIOLimit",
+                "supported_vol_types": ["generalPurpose"],
+                "title": "Percent IO Limit",
+            }
+        ]
+        widgets_list += self._add_conditional_storage_widgets(
+            conditional_metrics=conditional_efs_metrics_params,
+            volumes_list=efs_volumes_list,
+            namespace="AWS/EFS",
+            dimension_vol_name="FileSystemId",
+            attribute_name="file_system_id",
+            vol_attribute_name="performance_mode",
+        )
+
+        cloudwatch_dashboard.add_widgets(*widgets_list)
+        self._update_coord_after_section(self.coord, self.graph_height)
+
+    def _add_fsx_metrics_graphs(self, cloudwatch_dashboard):
+        self._add_text_widget("\n## FSx Metrics\n", cloudwatch_dashboard)
+        fsx_volumes_list = []  # TODO: Use the list of FSX in self.storage_resource
+        for storage in self.config.shared_storage:
+            if storage.shared_storage_type == SharedStorageType.FSX:
+                fsx_volumes_list.append(storage)
+
+        fsx_metrics = [
+            {
+                "metrics": ["DataReadOperations", "DataWriteOperations"],
+                "title": "Data Read/Write Ops",
+                "period": 300,
+            },
+            {
+                "metrics": ["DataReadBytes", "DataWriteBytes"],
+                "title": "Data Read/Write Bytes",
+                "period": 300,
+            },
+            {"metrics": ["FreeDataStorageCapacity"], "title": "Free Data Storage Capacity"},
+        ]
+        widgets_list = self._add_storage_widgets(
+            metrics=fsx_metrics,
+            storages_list=fsx_volumes_list,
+            namespace="AWS/FSX",
+            dimension_name="FileSystemId",
+            attribute_name="file_system_id",
+        )
+
+        cloudwatch_dashboard.add_widgets(*widgets_list)
+        self._update_coord_after_section(self.coord, self.graph_height)
+
+    def _add_cw_log(self, cloudwatch_dashboard):
+        self._add_text_widget("\n## Head Node Logs\n", cloudwatch_dashboard)
+
+        dcv_enabled = self.dcv_enable
+        scheduler = self.config.scheduling.scheduler
+        base_os = self.config.image.os
+        head_private_ip = self.head_node_instance.attr_private_ip
+        sections_widgets = [
+            {
+                "section_title": "ParallelCluster''s logs",
+                "widgets": [
```

Comment:
You can use namedtuples here too (e.g. CWLogWidget)

Reply:
Good idea! Done